clang-tools-extra/clangd/CodeComplete.cpp

```diff
@@ -2108,7 +2116,7 @@ class CodeCompleteFlow {
 
 } // namespace
 
-clang::CodeCompleteOptions CodeCompleteOptions::getClangCompleteOpts() const {
+clang::CodeCompleteOptions CodeCompleteOptions::getClangCompleteOpts(bool ForceLoadExternal) const {
```

Comment:
instead of taking a parameter can you make it part of `clangd::CodeCompleteOptions`  and initialize it to true if we have `EnableExperimentalModulesSupport` ?

Reply:
I found the `CodeCompleteOptions` are passing to `codeComplete`. So if we pass the information if modules are enabled when initialize `clangd::CodeCompleteOptions`, we may need to update all the places using `clangd::CodeCompleteOptions`. I feel it is bigger but unnecessary change. I don't feel bad with the current style. WDYT?

Reply:
> we may need to update all the places using clangd::CodeCompleteOptions

i don't follow this one, i'd expect us to just set it once in ClangdMain.cpp, with other CodeCompleteOpts fields we set based on command line flags/configs and use it only once in `CodeCompleteOptions::getClangCompleteOpts` (modulo tests that're specifically testing this behavior, not even all the tests).

What are the other places that need updating?

Reply:
It looks like we need to update `Check.cpp` too. My concern is majorly that `clangd::CodeCompleteOptions` may be used by someone else surprisingly. But given the development pattern in clangd, may be it is not a problem.

Reply:
Oh, it looks like you're right. Done.